app.py

```diff
@@ -49,19 +50,26 @@ def get_registration_page():
     """
     if request.method == 'POST':
         email_from_form = request.form.get('email')
-        if email_from_form:
+        password_from_form = request.form.get('password')
+        if not email_from_form or not password_from_form:
+            return 'You need all inputs to be filled in!', 400
+        else:
             connection = get_flask_database_connection(app)
             repository = UserRepository(connection)
-            if repository.check_email_exists(email_from_form) == False:
-                new_user = User(None, email_from_form) 
-                repository.create_new_user(new_user)
+            if repository.check_email_exists(email_from_form) == True:
+                return "Email has already been registered", 400
             else:
-                return "Email already exists"
-            
+                special_characters = '!@$%^&#~;:><=+-'
```

Comment:
This can be consigned to a helper method to keep the logic concise.

Reply:
Have moved to a user repository as a method to user including tests